packages/gatsby/src/query/file-parser.js

```diff
@@ -322,6 +323,73 @@ async function findGraphQLTags(
           },
         })
 
+        // Look for static images
+
+        // Generates a query document from the component props
+        const queryFromImage = (props, nodePath) => {
+          const text = queryFromProps(props)
+          const hash = hashFromQuery(text)
+          const fakeFile = `/STATICIMAGES/${hash}`
+
+          const gqlAst = parse(text)
+
+          gqlAst.definitions.forEach(def => {
+            generateQueryName({
+              def,
+              hash,
+              file: fakeFile,
+            })
+          })
+
+          const docInFile = {
+            filePath: fakeFile,
+            doc: gqlAst,
+            text,
+            hash,
+            isStaticQuery: true,
+            isHook: true,
+            templateLoc: nodePath.node.loc,
+          }
+
+          documentLocations.set(
+            docInFile,
+            `${nodePath.node.start}-${gqlAst.loc.start}`
+          )
+
+          documents.push(docInFile)
+        }
+
+        const componentImport = `StaticImage`
+        let localName = componentImport
+        traverse(ast, {
+          ImportSpecifier(path) {
+            if (path.node.imported.name === componentImport) {
```

Comment:
Is there a better way of doing this? I wasn't sure about the syntaxt of `referencesImport`, for example.